OpenTrons/opentrons-api
# feature(api, robot-server): Allow fixit commands to recover from an error

# Overview

closes https://opentrons.atlassian.net/browse/EXEC-285.
Allow FIXIT command intent when waiting for recovery and allow resume from recovery. 

# Test Plan

Tested on dev server works as expected!

- set error recovery ff to true `OT_API_FF_enableErrorRecoveryExperiments=1 `
- upload the following protocol - it should fail:
```
from opentrons import protocol_api

requirements = {"robotType": "Flex", "apiLevel": "2.17"}


def run(protocol: protocol_api.ProtocolContext) -> None:
    protocol.load_trash_bin("A3")

    tip_rack = protocol.load_labware("opentrons_flex_96_tiprack_50ul", "C2")
    reservoir = protocol.load_labware(
        "opentrons_96_wellplate_200ul_pcr_full_skirt", "C3"
    )
    pipette = protocol.load_instrument(
        "flex_1channel_50", mount="left", tip_racks=[tip_rack]
    )

    for source, dest in zip(reservoir.columns()[0], reservoir.columns()[1]):
        pipette.pick_up_tip()
        pipette.move_to(source.top())
        pipette.move_to(dest.top())
        pipette.drop_tip()
```
- GET `/run` and make sure it is in `AWAITING_RECOVERY` mode. 
- POST a few commands with `intent=FIXIT`
- make sure commands are executed.
- when done, issue a resumeFromRecovery action.  # need to check if this is true


# Review requests

Changes make sense? did I miss anything?

# Risk assessment

Medium. changing `add_command` in pe and `create_command` in router.
